packages/builder/src/components/design/settings/controls/ButtonActionEditor/actions/DownloadFile.svelte

```diff
@@ -0,0 +1,76 @@
+<script>
+  import { Select, Label } from "@budibase/bbui"
+  import { onMount } from "svelte"
+  import DrawerBindableInput from "components/common/bindings/DrawerBindableInput.svelte"
+  import { FieldType } from "@budibase/types"
+
+  export let parameters
+  export let bindings = []
+
+  $: fileBindings = bindings?.filter(
+    b => b.fieldSchema?.type === FieldType.ATTACHMENT
+  )
+
+  const fileOptions = [
+    {
+      label: "Attachment",
+      value: "attachment",
+    },
+    {
+      label: "Url",
+      value: "url",
+    },
+  ]
+
+  onMount(() => {
+    if (!parameters.type) {
+      parameters.type = "attachment"
+    }
+  })
+</script>
+
+<div class="root">
+  <Label small>File</Label>
+  <Select
+    placeholder={null}
+    bind:value={parameters.type}
+    options={fileOptions}
+  />
+  {#if parameters.type === "attachment"}
+    <Label small>Attachment</Label>
+    <DrawerBindableInput
+      title="Attachment"
+      bindings={fileBindings}
+      allowHelpers={false}
+      value={parameters.attachment}
+      on:change={value => (parameters.attachment = value.detail)}
+    />
+  {:else}
+    <Label small>URL</Label>
+    <DrawerBindableInput
+      title="URL"
+      {bindings}
+      value={parameters.url}
+      on:change={value => (parameters.url = value.detail)}
+    />
+    <Label small>File name</Label>
+    <DrawerBindableInput
+      title="File name"
+      {bindings}
+      value={parameters.file_name}
```

Comment:
NAB: you don't need to change this, and we never enforce it, but by convention we always just use camel case for variable naming in the frontend. Just an FYI for the future!

Reply:
I will check if we can add a lint rule without breaking the existing usages :)

Reply:
Done!